backend/schemaservice/schemaservice.go

```diff
@@ -68,34 +71,49 @@ func Start(
 	logger.Debugf("Starting FTL schema service")
 
 	g, gctx := errgroup.WithContext(ctx)
+	gctx, shutdown := context.WithCancelCause(gctx)
+
+	var rpcOpts []rpc.Option
 
 	var shard statemachine.Handle[struct{}, SchemaState, EventWrapper]
 	if config.Raft.DataDir == "" {
 		// in local dev mode, use an inmemory state machine
 		shard = statemachine.NewLocalHandle(newStateMachine(ctx))
 	} else {
 		clusterBuilder := raft.NewBuilder(&config.Raft)
-		schemaShard := raft.AddShard(ctx, clusterBuilder, 1, newStateMachine(ctx))
-		cluster := clusterBuilder.Build(ctx)
-		if err := cluster.Start(ctx); err != nil {
-			return fmt.Errorf("failed to start raft cluster: %w", err)
-		}
+		schemaShard := raft.AddShard(gctx, clusterBuilder, 1, newStateMachine(ctx))
 		shard = schemaShard
+
+		rpcOpts = append(rpcOpts, raft.RPCOption(clusterBuilder))
 	}
 
 	svc := New(ctx, shard, config)
 	logger.Debugf("Listening on %s", config.Bind)
 
 	g.Go(func() error {
 		return rpc.Serve(gctx, config.Bind,
-			rpc.GRPC(ftlv1connect.NewSchemaServiceHandler, svc),
-			rpc.PProf(),
+			append(rpcOpts,
+				rpc.GRPC(ftlv1connect.NewSchemaServiceHandler, svc),
+				rpc.PProf(),
+			)...,
 		)
 	})
 
+	// Handling graceful shutdown.
+	termSignalChannel := make(chan os.Signal, 1)
```

Comment:
I think all of this (and the WithCancelCause) can be replaced with `signal.NotifyContext()` no?

Reply:
Ah, nice! Yes